packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -144,47 +148,39 @@ contract MixinExchangeCore is
 
     /// @dev Cancels the input order.
     /// @param order Order struct containing order specifications.
-    /// @param takerTokenCancelAmount Desired amount of takerToken to cancel in order.
-    /// @return Amount of takerToken cancelled.
-    function cancelOrder(
-        Order order,
-        uint256 takerTokenCancelAmount)
+    /// @return Success of cancel.
+    function cancelOrder(Order order)
         public
-        returns (uint256 takerTokenCancelledAmount)
+        returns (bool)
     {
         // Compute the order hash
         bytes32 orderHash = getOrderHash(order);
         
         // Validate the order
         require(order.makerTokenAmount > 0);
         require(order.takerTokenAmount > 0);
-        require(takerTokenCancelAmount > 0);
         require(order.makerAddress == msg.sender);
 
         if (block.timestamp >= order.expirationTimeSeconds) {
             LogError(uint8(Errors.ORDER_EXPIRED), orderHash);
-            return 0;
+            return false;
         }
 
-        uint256 remainingTakerTokenAmount = safeSub(order.takerTokenAmount, getUnavailableTakerTokenAmount(orderHash));
-        takerTokenCancelledAmount = min256(takerTokenCancelAmount, remainingTakerTokenAmount);
-        if (takerTokenCancelledAmount == 0) {
-            LogError(uint8(Errors.ORDER_FULLY_FILLED_OR_CANCELLED), orderHash);
-            return 0;
+        if (cancelled[orderHash]) {
+            LogError(uint8(Errors.ORDER_CANCELLED), orderHash);
+            return false;
         }
```

Comment:
I'm trying to understand what a return value `true` means. It doesn't mean "the order is unfillable", because we return false when the order is expired or already cancelled. Besides, after calling cancel the order is always unfillable.

It could mean "the order changed from fillable to unfillable". But then we also need to handle `Errors.ORDER_FULLY_FILLED`. This would require an extra state read though.

Reply:
As is it's "the order changed from not cancelled to cancelled". Definitely a bit awkward though.

Reply:
Maybe we can adjust the docs to read:

```
/// After calling, the order can not be filled anymore.
/// @returns true if the order state changed to Cancelled. false if the transaction was already cancelled or othwerise unfillable.
```

Success just seems a bit confusing, this call never fails in the sense of taking an order down.